### ISSUE TITLE
Magnet links defaulting to wrong download location

### ISSUE THREAD

**Issue by 0jay** [transmission-remote-gui/transgui#1155]

**Comment by leonsoft-kras** [comment]
What is the issue or problem?

**Comment by 0jay** [comment]
I'm unable to click magnet links to activate torrents as it results in a flase download location which is different to how it's configured either on the router or in the app

**Comment by leonsoft-kras** [comment]
It is right. You added a magnet link, and your transmission has not yet downloaded the file list.

**Comment by 0jay** [comment]
When I paste a magnet link or upload a torrent file the torrent downloads to the correct location.

Only when activating the app by clicking a magnet link does it the error out because the location it defaults to does not exist

**Comment by leonsoft-kras** [comment]
![default](https://user-images.githubusercontent.com/11468577/43991786-fdb6e496-9d9d-11e8-82bd-3c96d25a99db.png)

Do you see such a window when adding a torrent?

**Comment by 0jay** [comment]
No, I have it set so as to begin automatically therefore it defaults to a download location rather than an option to choose the download location.

The problem is the location it is defaulting to.

**Comment by leonsoft-kras** [comment]
OK. It seems understood. It seems someone was engaged in this topic.

**Comment by 0jay** [comment]
Thanks for your time mate.

**Comment by rzcat** [comment]
I cannot re-preoduce the problem. I got something wrong?
With the checkbox "Prompt for download options when adding a new torrent" CHECKED, when I click a magnet link, the "Add a new torrent" dialogue appears with my pre-set Destination folder
With the checkbox UNCHECKED, when I click a magnet link, the task will be added to my queue list, with the Destination folder set as if the "Add a new torrent" dialogue been clicked "OK" automatically

**Comment by 0jay** [comment]
Hi, ta for looking into it.

The thing I'd mention that I think has caused this is that the directory it defaults to is a directory I inputed into your gui before I'd worked out how to get the attached USB drive to work so it's a location that I had previously inputed. The new (correct) location has been inputed into the gui and into the config file in the command line and yet it still defaults to the previous location. 

I've tried deleting Transmission Remote GUI with AppCleaner, removing everything yet on reinstalling the preferences are saved.

Thanks for your time. 
Oscar

**Comment by 0jay** [comment]
Solved.

Had to reenable the prompt for downloads option to delete previous locations.

Thanks for your app - I love it!

**Comment by anohren** [comment]
I have the same issue.

**Comment by miroesq** [comment]
I am experiencing the same thing. I have transmission daemon installed on a linux box and am using transmission qt client on windows to log into it remotely. I have the default file save location as "/mnt/drive" and when I add a torrent file, there are no problems. However, when I click on a magnet link in my browser on windows, I get a transmission popup and the default file location is "C:/mnt/drive" which is wrong and I have to change it manually every time which is super annoying. I've uninstalled and reinstalled chrome, and windows transmission qt, but still it persists.

**Comment by anohren** [comment]
0jay hinted at a workaround, but this doesn't seem to get a fix any time soon so here's a more detailed alternative workaround:

Search for `AddTorrent` in `~/.config/Transmission Remote GUI/transgui.ini` and edit all paths that match the one you don't want under the sections called "AddTorrent.<your server name>`. There's one set of settings per server.

Uninstalling the application often leaves configuration files behind which would explain why that didn't work.

The application is trying to suggest the last used directory that you downloaded to, which is all fine and convenient _if you've enabled the option to show the download options dialog when a torrent is added_. If that option isn't enabled then it backfires by still defaulting to the last manually used directory, instead of the actual default download directory.

In other words, you lock yourself into the most recently used destination folder as soon as you disable the download options dialog for new transfers.
